odo crashed when running  catalog list components

/kind bug

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the Google group if you have a question rather than a bug or feature request.

The group is at: https://groups.google.com/forum/#!forum/odo-users

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:**

[root@xyimgr77 ~]# ./odo version
odo v1.1.0 (86c7e2c9)

Server: https://api.ocp-test.redhat.com:6443
Kubernetes: v1.14.6+8bbaf43
## How did you run odo exactly?

create 3 image streams, the 2nd and 3rd might share some same info, but can't remember what's changed.. lead to following crash

```
[root@xyimgr77 frontend]# ../odo catalog list components
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/openshift/odo/pkg/catalog.createImageTagMap(0xc0000b2370, 0x1, 0x1, 0xc000628ea0)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/catalog/catalog.go:263 +0x37e
github.com/openshift/odo/pkg/catalog.SliceSupportedTags(0x15d2b0a, 0xd, 0x15e340a, 0x19, 0xc0007dffd0, 0x7, 0x0, 0x0, 0xc0007dffe0, 0x9, ...)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/catalog/catalog.go:230 +0x86
github.com/openshift/odo/pkg/odo/cli/catalog/list.(*ListComponentsOptions).Run(0xc000417e00, 0x0, 0x0)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/catalog/list/components.go:72 +0x2c4
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x18a3340, 0xc000417e00, 0xc000a27b00, 0x2a4ac90, 0x0, 0x0)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:31 +0x168
github.com/openshift/odo/pkg/odo/cli/catalog/list.NewCmdCatalogListComponents.func1(0xc000a27b00, 0x2a4ac90, 0x0, 0x0)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/catalog/list/components.go:112 +0x4e
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc000a27b00, 0x2a4ac90, 0x0, 0x0, 0xc000a27b00, 0x2a4ac90)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:702 +0x26c
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000a3c900, 0x16d21e8, 0xc000087ce0, 0x0)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:783 +0x28e
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:736
main.main()
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:72 +0x42e
```
## Actual behavior


## Expected behavior


## Any logs, error output, etc?

--- a/pkg/catalog/catalog.go
+++ b/pkg/catalog/catalog.go
@@ -260,7 +260,17 @@ func createImageTagMap(tagRefs []imagev1.TagReference) map[string]string {
 		imageName := tagRef.From.Name
 		if tagRef.From.Kind == "DockerImage" {
 			// we get the image name from the repo url e.g. registry.redhat.com/openshift/nodejs:10 will give openshift/nodejs:10
-			urlImageName := strings.SplitN(imageName, "/", 2)[1]
+			imageNameParts := strings.SplitN(imageName, "/", 2)
+
+			var urlImageName string
+			// this means the docker image url might just be something like nodejs:10, no namespace or registry info
+			if len(imageNameParts) == 1 {
+				urlImageName = imageNameParts[0]
+				// else block executes when there is a registry information attached in the docker image url
+			} else {
+				// we dont want the registry url portion
+				urlImageName = imageNameParts[1]
+			}
 			// here we remove the tag and digest
 			ns, img, tag, _, _ := occlient.ParseImageName(urlImageName)
 			imageName = ns + "/" + img + ":" + tag

